main.c

```diff
@@ -170,10 +170,23 @@ void fetch_one_suite(const char * suite_path, struct application_context * appli
 
 void read_suites(struct application_context * application_context)
 {
+    /* replacing a standard array with something more appropriate like a hashmap is probably what we need but this should do for now. */
+    const char * fnames[1024];
```

Comment:
> Hi, thank you for a contribution, but i see some problems. Could you fix it?
> 
> The problems:
> 
>     1. The code tries to follow single responsibility concept. And i think that kind of fix needs to be place in the `fetch_one_suite` instead of `read_suites`, because it's a "fetching" problem not "reading". The `read_suites` responsibility are just decide `read the suite from some kind of cache` or `just compile it from the scratch`. And yeah, I agree that the implementation should be a hashmap or a binary tree, I plan to add this in the future. Or if you want you could try to do this by yourself. (the project for now using a static allocators in the `allocate.c`)
> 
>     2. Some review problems.

I agree, yet doing it before their compilation was much simpler, and doesn't really affect performance either. I will try to read it again and see if i can find a way to include the check in fetch_suite, will also follow your desired name conventions.